src/CrudPanel.php

```diff
@@ -237,16 +239,91 @@ public function sort($items)
      *          App/Models/Address defined by a company() method on the user model and an address() method on the
      *          company model, the 'App/Models/Address' string will be returned.
      *
-     * @param $relationString String Relation string. A dot notation can be used to chain multiple relations.
+     * @param string $relationString Relation string. A dot notation can be used to chain multiple relations.
+     * @param int $length Optionally specify the number of relations to omit from the start of the relation string. If
+     *        the provided length is negative, then that many relations will be omitted from the end of the relation
+     *        string.
+     * @param Model $model Optionally specify a different model than the one in the crud object.
      *
-     * @return string relation model name
+     * @return string Relation model name.
      */
-    private function getRelationModel($relationString)
+    public function getRelationModel($relationString, $length = null, $model = null)
     {
-        $result = array_reduce(explode('.', $relationString), function ($obj, $method) {
+        $relationArray = explode('.', $relationString);
+
+        if (! isset($length)) {
+            $length = count($relationArray);
+        }
+
+        if (! isset($model)) {
```

Comment:
Should `$model` subclass something? What if it is, say, an integer (i.e. the literal number `2`) or an `App\Http\Kernel` or something silly? Then again see my above comment re: silly inputs.

Reply:
The model should be an Eloquent model as specified. If it is not, it will obviously fail with an error.

Even though it's specified as only "Model", PhpStorm adds the correct class to the import statements and does not use the fully qualified name. This works for me, but it might be harder to interpret when viewing on GitHub or other IDEs. 

Would you think that having something like `@param Illuminate\Database\Eloquent\Model $model` will be more readable or would end up cluttering the phpdoc even more?

Reply:
It should be the full name! PHP Storm costs people who make no money...money.